lib/quby/table_backend/table_dimension.rb

```diff
@@ -0,0 +1,42 @@
+module Quby::TableBackend
+# TableDimension acts like nodes of a tree, every level consists of an array of table dimensions that each
+# have multiple ranges that point to either an array of TableDimensions for the next dimension or
+#   a non-TableDimension, which will be seen as a result
+
+# Going down the tree a level is like evaluating dimensions AND-wise,
+# going side to side on a level is evaluating dimensions OR-wise
+  class TableDimension < Struct.new(:name, :ranges)
```

Comment:
Dat het een filename is, is een concern van DiskTable. De TableDimension maalt niet waar de namen vandaan komen. De structuur van de ranges mag inderdaad wel wat verduidelijking.